lib/beaker/test_suite_result.rb

```diff
@@ -150,6 +150,25 @@ def print_test_result(test_case)
       "  Test Case #{test_case.path} #{test_reported}"
     end
 
+    # Saves failure and error cases as an options file for only-failures processing
+    #
+    # @param [Boolean] only_failures    Are we rerunning with fails/errors?
+    # @param [String] save_results_file Where to put the results if so
+    #
+    def persist_results(only_failures, save_results_file)
+      unless save_results_file.empty?
```

Comment:
```suggestion
      return if save_results_file.empty?
```
in a situation like this where the conditional is really just guarding execution, we'd prefer to use a guard clause statement like the suggestion rather than the body as is, since it keeps the method less nested.

I don't think applying the suggestion will work as-is though, since the `end` statement at the end of the method will need deleting & most of the method will need unindenting a bit.

Note that this applies to the `unless only_failures` conditional just below this as well.

Reply:
No, I get it. Currently waiting on Enis to complete work on porting PA+friends to Beaker 4.